### PR TITLE
[SCSB-155] build with Numpy 2.0 release candidate

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-155](https://jira.stsci.edu/browse/SCSB-155)

Numpy 2.0 is scheduled for release on June 19th. Due to the C ABI updates, Python projects that build C extensions are required to build against Numpy 2.0 in order to use Numpy 2.0 in runtime.
This is backwards-compatible with Numpy versions in runtime as far back as `1.26`

this PR pins `numpy>=2.0.0rc2` in `build-system.requires`
> [!NOTE]
> this PR was generated automatically by ``batchpr`` :robot:

**Checklist for maintainers**
- [ ] added entry in `CHANGELOG.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,89 +1,107 @@
...
-    'tables',
-    'typing_extensions>4.0',
-    'markupsafe<=2.0.1',
+    "astropy>=5.0.4",
+    "fitsblender>=0.4.2",
+    "scipy",
+    "matplotlib",
+    "stsci.tools>=4.0",
+    "stsci.image>=2.3.7",
+    "stsci.imagestats",
```
Comment (mdlpstsci) [inline pyproject.toml:33]: We want to make sure drizzlepac uses stsci.imagestats >= 1.8.2 (**_NOT_** 1.7.0, 1.8.0, or 1.81. though I see these are currently "yanked").

File: pyproject.toml [hunk 2]
```diff
@@ -1,89 +1,107 @@
...
+    "stsci.skypac>=1.0.9",
+    "stsci.stimage",
+    "stwcs>=1.5.3",
+    "tweakwcs>=0.8.2",
```
Comment (mdlpstsci) [inline pyproject.toml:37]: We want to make sure drizzlepac uses tweakwcs >= 0.8.7 as there is a bug in 0.8.6.

File: pyproject.toml [hunk 3]
```diff
@@ -95,90 +113,88 @@ zip-safe = false
...
-    'F521', # '...'.format(...) has invalid format string: Single '}' encountered in format string
-    'F523', # '...'.format(...) has unused arguments at position(s): 0, 1
-    'F541', # f-string without any placeholders
-    'F632', # Use `==` and `!=` to compare constant literals
-    'F633', # Use of `>>` is invalid with `print` function
-    'F811', # Redefinition of unused
-    'F821', # Undefined name
-    'F822', # Undefined name `getHelpAsString` in `__all__`
-    'F841', # Local variable is assigned to but never used
+    "E401",
```
Comment (mdlpstsci) [inline pyproject.toml:179]: Just a question here.  Is it against the new rules to have the commentary associated with the error codes to ignore.  The commentary is rather helpful.  Thanks.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
Some changes need to be made to ensure drizzlepac is using "good" versions of the stsci.imagestats and tweakwcs packages.  I also had one question.

File: pyproject.toml (hunk 3 above)
Comment (zacharyburnett) [reply pyproject.toml:179]: oh, sorry! That must have been the formatter; I'll add those comments back in

**Review by mdlpstsci — APPROVED** [review]
Thanks for addressing these issues. You could not have known about the imagestats and tweakwcs versions!

**Pull request merged**